examples/honest.rs

```diff
@@ -0,0 +1,323 @@
+use futures::{
+    channel::{
+        mpsc::{self, UnboundedReceiver, UnboundedSender},
+        oneshot,
+    },
+    Future, FutureExt, StreamExt,
+};
+use log::info;
+use parking_lot::Mutex;
+use rush::{NodeIndex, OrderedBatch};
+use std::{
+    collections::{hash_map::DefaultHasher, HashSet},
+    error::Error,
+    hash::Hasher as StdHasher,
+    sync::Arc,
+};
+use tokio::time::Duration;
+
+use libp2p::{
+    development_transport,
+    floodsub::{self, Floodsub, FloodsubEvent},
+    identity,
+    mdns::{Mdns, MdnsConfig, MdnsEvent},
+    swarm::{NetworkBehaviourEventProcess, SwarmBuilder},
+    NetworkBehaviour, PeerId, Swarm,
+};
+
+const USAGE_MSG: &str = "Missing arg. Usage
+    cargo run --example honest my_id n_members n_finalized
+
+    my_id -- our index
+    n_members -- size of the committee
+    n_finalized -- number of data to be finalized";
+
+fn parse_arg(n: usize) -> usize {
+    if let Some(int) = std::env::args().nth(n) {
+        match int.parse::<usize>() {
+            Ok(int) => int,
+            Err(err) => {
+                panic!("Failed to parse arg {:?}", err);
+            }
+        }
+    } else {
+        panic!("{}", USAGE_MSG);
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::builder()
+        .filter_module("honest", log::LevelFilter::Info)
+        .init();
+
+    let my_id = parse_arg(1);
+    let n_members = parse_arg(2);
+    let n_finalized = parse_arg(3);
+
+    info!("Getting network up.");
+    let (network, mut manager) = Network::new().await.unwrap();
+    let (close_network, exit) = oneshot::channel();
+    tokio::spawn(async move { manager.run(exit).await });
+
+    let (data_io, mut to_finalize) = DataIO::new();
+
+    let mut interval = tokio::time::interval(Duration::from_millis(500));
+    interval.tick().await;
+
+    while network.connected_peers() != n_members - 1 {
+        info!(
+            "waiting for peers, missing {}",
+            n_members - network.connected_peers() - 1
+        );
+        interval.tick().await;
+    }
+
+    let (close_member, exit) = oneshot::channel();
+    tokio::spawn(async move {
+        let keybox = KeyBox {
+            index: my_id.into(),
+        };
+        let config = rush::default_config(n_members.into(), my_id.into(), 0);
+        let member = rush::Member::new(data_io, &keybox, config);
+
+        member.run_session(network, Spawner {}, exit).await
+    });
+
+    let mut finalized = HashSet::new();
+    while let Some(batch) = to_finalize.next().await {
+        for data in batch {
+            if !finalized.contains(&data) {
+                finalized.insert(data);
+            }
+        }
+        info!("Got new batch. Finalized = {:?}", finalized.len());
+        if finalized.len() == n_finalized {
+            break;
+        }
+    }
+    close_member.send(()).expect("should send");
+    close_network.send(()).expect("should send");
+}
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+struct Hasher64;
```

Comment:
I doubt this is cryptographically secure.

Reply:
I wasn't suppose to be, adding comment.